src/service/Offers.ts

```diff
@@ -108,55 +109,52 @@ export const getOfferService = (
         ),
       );
 
-    const [
-      instance,
-      give,
-      want,
-      displayableGiveTemplate,
-      displayableWantTemplate,
-    ] = await Promise.all([
-      E(boardIdMarshaller).unserialize(instanceHandle),
-      convertProposals(giveTemplate),
-      convertProposals(wantTemplate),
-      makeProposalTemplateDisplayable(giveTemplate),
-      makeProposalTemplateDisplayable(wantTemplate),
-    ]);
+    const [give, want, displayableGiveTemplate, displayableWantTemplate] =
+      await Promise.all([
+        convertProposals(giveTemplate),
+        convertProposals(wantTemplate),
+        makeProposalTemplateDisplayable(giveTemplate),
+        makeProposalTemplateDisplayable(wantTemplate),
+      ]);
 
-    const offerForAction: OfferSpec = {
-      id,
-      invitationSpec: {
+    let invitationSpecToUse = invitationSpec;
+    let instanceBoardId = 'Continuing Invitation';
+
+    if (!invitationSpecToUse) {
+      const instance = await E(boardIdMarshaller).unserialize(instanceHandle);
+      invitationSpecToUse = {
         source: 'contract',
         instance,
         publicInvitationMaker,
-      },
+      };
+      instanceBoardId = `instance@${
+        (await E(boardIdMarshaller).serialize(instance)).slots[0]
+      }`;
+    }
+
+    const offerForAction: OfferSpec = {
+      id,
+      invitationSpec: invitationSpecToUse,
       proposal: {
         give,
         want,
       },
     };
 
-    const [
-      {
-        slots: [instanceBoardId],
-      },
-      spendAction,
-    ] = await Promise.all([
-      E(boardIdMarshaller).serialize(instance),
-      E(boardIdMarshaller).serialize(
-        harden({
-          method: 'executeOffer',
-          offer: offerForAction,
-        }),
-      ),
-    ]);
+    const spendAction = await E(boardIdMarshaller).serialize(
+      harden({
+        method: 'executeOffer',
+        offer: offerForAction,
+      }),
+    );
 
     return {
       ...offer,
       proposalTemplate: {
         give: displayableGiveTemplate,
         want: displayableWantTemplate,
       },
-      instancePetname: `instance@${instanceBoardId}`,
+      instancePetname: instanceBoardId,
```

Comment:
this is no longer about an instance. Looking at where this value is used, maybe it's something like `sourceDescription`?